### PR TITLE
check for max and min in MTU

### PR DESCRIPTION
**Pull request by Behnam-Shobiri**
## Description
Fix the string to int conversion. Add upper bound check for converting strings to int. 

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/installation/core_controller.go
Comment (tmjd) [inline pkg/controller/installation/core_controller.go:1369]: Instead of checking greater than MinInt32 we should just check for >= 0 because a negative MTU is invalid.

Comment (Behnam-Shobiri) [reply pkg/controller/installation/core_controller.go]: Fixed

**Review by tmjd — APPROVED** [review]
LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
